docs/pages/management/guides/joining-services-kubernetes-serviceaccount.mdx

```diff
@@ -0,0 +1,275 @@
+
+It is not recommended to use a single token that can join everything. You should
+restrict the token to the roles used by the joining instance. For example, a Teleport
+instance running both the Application Service and Database Service should use a token with
+`roles: [App, Db]`.
+Follow the instructions below to create the token depending on whether you have administrative access to the Auth Service pod:
+<Tabs>
+<TabItem label="Using tctl locally as your current user">
+Make sure your local `tctl` is at least at version 12 with `tctl version`.
+
+Create the token:
+
+```code
+$ tctl create token.yaml
+```
+
+Finally, validate the token was created:
+
+```code
+$ tctl get token/kubernetes-token
+
+kind: token
+metadata:
+  expires: "3000-01-01T00:00:00Z"
+  name: kubernetes-token
+spec:
+  join_method: kubernetes
+  roles:
+  - App
+version: v2
+```
+
+</TabItem>
+<TabItem label="Using tctl on the Auth Service as an admin">
+Retrieve the name and namespace of the Auth Service deployment:
+
+```code
+$ kubectl get namespaces
+NAME              STATUS   AGE
+cert-manager      Active   40d
+default           Active   40d
+kube-system       Active   40d
+teleport          Active   40d
+
+# We look for deployments in the "teleport" namespace
+$ kubectl get deployments -n teleport
+NAME             READY   UP-TO-DATE   AVAILABLE   AGE
+teleport-auth    2/2     2            2           6d20h
+teleport-proxy   2/2     2            2           6d20h
+
+# Here, the deployment name is "teleport-auth".
+```
+
+Then run the following command to execute the `tctl create` command from inside
+one of the Auth Service pods:
+
+```code
+$ kubectl exec -i -n teleport deployment/teleport-auth -- tctl create < token.yaml
+```
+
+Finally, validate the token was successfully created:
+
+```code
+$ kubectl exec -i -n teleport deployment/teleport-auth tctl get token/kubernetes-token
+
+kind: token
+metadata:
+  expires: "3000-01-01T00:00:00Z"
+  name: kubernetes-token
+spec:
+  join_method: kubernetes
+  roles:
+  - App
+version: v2
+```
+
+</TabItem>
+</Tabs>
+
+## Step 2/5. Deploy a demonstration HTTP app
+
+In this step, we deploy a demonstration HTTP application and don't expose it
+publicly. Instead, we will manage access to this application with the Teleport
+Application Service, which we will register with Teleport using the Kubernetes
+join method.
+
+```code
+$ kubectl create namespace demo-app
+namespace/demo-app created
+
+$ kubectl create deployment --image=nginx --namespace demo-app --port=80 demo-app
+deployment.apps/demo-app created
+
+$ kubectl expose deployment demo-app -n demo-app --port=80 --target-port=80 --selector='app=demo-app'
+service/demo-app exposed
+```
+
+Validate the application pods are running and ready with the following command:
+
+```code
+$ kubectl get pods -n demo-app
+NAME                        READY   STATUS    RESTARTS   AGE
+demo-app-7664d59cb8-bvbmz   1/1     Running   0          67s
```

Comment:
@hugoShaka  Can you update this to `bv888` as done in the master pr https://github.com/gravitational/teleport/pull/24553.  That will allow us not to have to add to the spell check ignore.

Reply:
👍 I cherry-picked your commit